python/kserve/kserve/storage.py

```diff
@@ -181,7 +181,7 @@ def _download_s3(uri, temp_dir: str):
             target_key = (
                 obj.key.rsplit("/", 1)[-1]
                 if bucket_path == obj.key
-                else obj.key.replace(bucket_path, "", 1).lstrip("/")
+                else obj.key.replace(bucket_path.rsplit("/",1)[0], "", 1).lstrip("/")
```

Comment:
Can you help add a test ? Thanks !

Reply:
@yuzisun I have added it, thank you.

Reply:
Thanks ! there is a small lint issue and please also help update the PR with meaningful title as that goes into the release notes.